hack/generator/pkg/codegen/pipeline/flatten_properties.go

```diff
@@ -52,7 +53,8 @@ func makeFlatteningVisitor(defs astmodel.Types) astmodel.TypeVisitor {
 
 			// safety check:
 			if err := checkForDuplicateNames(newProps); err != nil {
-				return nil, err
+				klog.Warningf("Flattening caused duplicate property names, skipping flattening")
```

Comment:
Maybe log the error here since that includes the property name?

Reply:
good point